src/secrets/CommandEnv.ts

```diff
@@ -33,7 +35,23 @@ class CommandEnv extends CommandPolykey {
         args: [Array<[string, string?, string?]>, Array<string>],
         options,
       ) => {
-        args[1].shift();
+        // The parser sets the default value for args. If no arguments are
+        // provided, then args is an empty array []. This is different from the
+        // expected structure. This block ensure that the structure is preserved.
+        // Do the same for options.preserveNewline.
+        args = args[0] == null && args[1] == null ? [[], []] : args;
+        options.preserveNewline = options.preserveNewline ?? [];
+
+        // There needs to be at least one argument or preserved argument
+        // provided. Otherwise, print the help text and error out.
+        if (args[0].length === 0 && options.preserveNewline.length === 0) {
+          this.outputHelp()
+          throw new commander.InvalidArgumentError(
+            'You must provide at least one secret path.',
+          );
+        }
```

Comment:
I don't like doing it this way, but I don't know any other way to make this work. I have set the newline preservation as an option. This means that we have zero secrets provided only when both `args` and `options.preserveNewline` are zero.

If there were no args, then the array will not get properly initialised. The same is the case for `options.preserveNewline`, so I had to manually create an empty array for both.

Then, `this.help()` will print the help page and exit with an error code of zero, so that can't be used for printing help text for an error, and the help text isn't printed for errors thrown from the action. It is only printed when it is thrown in the parsing step. So, I have to manually output the help page, and exit with the error so the correct error code gets set.

This approach just doesn't sit well with me, and there is got to be a better method to do this.

Reply:
Answered in https://github.com/MatrixAI/Polykey-CLI/pull/329#issuecomment-2472332972